src/web/gwc/src/main/java/org/geoserver/gwc/web/layer/UnconfiguredCachedLayersProvider.java

```diff
@@ -62,6 +75,80 @@ public int compare(TileLayer o1, TileLayer o2) {
     static final List<Property<TileLayer>> PROPERTIES =
             Collections.unmodifiableList(Arrays.asList(TYPE, NAME, ENABLED));
 
+    private static final String KEY_SIZE = "key.size";
+    private final Cache<String, Integer> cache;
+
+    private GWCConfig defaults;
+
+    public UnconfiguredCachedLayersProvider() {
+        super();
+        // Initialization of an inner cache in order to avoid to calculate several times
+        // the size() method in a time minor than a second
+        CacheBuilder<Object, Object> builder = CacheBuilder.newBuilder();
+        cache = builder.expireAfterWrite(1, TimeUnit.SECONDS).build();
```

Comment:
I concur. I've copied the approach from `PreviewLayerProvider` where it's used to avoid calling `size()` and `fullSize()` repeatedly during the same page load (for this case, `size()` is called 5 times and `fullSize()` once as far as I could see).
Key'ing the cache by `filter` (all standard filter implementations do implement equals afaik) and having a max size of 1 would ensure correctness too (provided it's only used for `size()` and not for `fullSize()`). Agreed?

Reply:
they don't, but we could save the filter terms as a set instead.

Reply:
Yeah, keying by keywords should work. But for the sake of the caching, you can just keep the filter in a transient field, and re-compute the size if missing.